core/src/wsv.rs

```diff
@@ -739,6 +804,9 @@ pub mod config {
         pub domain_metadata_limits: MetadataLimits,
         /// [`LengthLimits`] for the number of chars in identifiers that can be stored in the WSV.
         pub ident_length_limits: LengthLimits,
+        /// Amount of time Peer waits for CommitMessage from the proxy tail.
+        /// Should have the same value as corresponding field in sumeragi config
+        pub commit_time_ms: u64,
```

Comment:
Yeah, but I haven't find the way, to unify it. Any suggestions?

Reply:
One way may be to stop using time ranges altogether.
Just check if the appointed time is after the block timestamp and disable it after the execution in one-shot cases, for example.
I think I saw that in your previous implementation though

Reply:
If I'll check if appointed time is after the block timestamp, then all future triggers will be executed immediately, event if they should be executed in a year from now. That's not what we want.

In my previous realization I checked if appointed time is before the block time. In that way time-triggers always will be executed with late. Aleksander assumed, that there can be some cases, then consensus takes too long (from time-triggers point of view), so we decided to use this approach and run triggers before...

We need @appetrosyan  here for him to explain more details

Reply:
Yeah I got the before and the after wrong, sorry

Reply:
Basically, here's the deal. We can choose which point in the consensus process amounts to the fairest execution policy. 

Now suppose you have a block commit time of 1 hour, +/- 30 minutes. Triggers will aim to fit into that time frame. Some will be late, some will be early. 

1) if we use the beginning of the commit process (the block timestamp), only the triggers that were accidentally faster will be executed in this block, but things that were even a nanosecond late will have to wait until the next block. 

2) we could project a time later than the block timestamp that we estimate is the time when consensus is reached. This is the current implementation, and one I advised @Arjentix to consider. The main reasoning is that the latency is ever so slightly lower. More importantly, the triggers that get executed shall be symmetrically distributed on both sides of the commit time. 

Since we do "best effort", we want to encourage people to aim as close to block commit time, rather than before block commit time.

Reply:
If we want to get mathematical, we should mention the central limit theorem, and the fact that an asymmetrical distribution is not separable, thus the error in block commit time can never be regarded as a normal distribution. 

This would make any kind of data science on timed triggers way harder to do. For example, you wouldn't be able to extrapolate further into the future how many times a given trigger will run. It will always be +/- one time. In a Gaussian distribution the error margin can become smaller based on the number of repetitions.

Reply:
Either one of the times works. 

Ideally we want to have a linear combination of both, and tune it so we have the smallest number of triggers that get cut-off.

Reply:
I'm closing this conversation, cause the original topic was about `commit_time_ms` which is deleted from *WSV* config now